src/vs/workbench/api/node/extHostTypes.ts

```diff
@@ -1754,17 +1785,33 @@ export class Task implements vscode.Task {
 	}
 
 	get execution(): ProcessExecution | ShellExecution | undefined {
-		return this._execution;
+		return (this._execution instanceof ExtensionCallbackExecution) ? undefined : this._execution;
 	}
 
 	set execution(value: ProcessExecution | ShellExecution | undefined) {
+		this.executionInternal = value;
+	}
+
```

Comment:
When the API is finalized ideally we'd just be able to delete the old `execution` getter/setter and rename the `executionWithExtensionCallback` getter/setter to `execution` and not have any other cleanupto do here.